pyqtgraph/widgets/MultiAxisPlotWidget.py

```diff
@@ -0,0 +1,402 @@
+__all__ = ["MultiAxisPlotWidget"]
+
+import weakref
+
+from ..graphicsItems.AxisItem import AxisItem
+from ..graphicsItems.PlotDataItem import PlotDataItem
+from ..graphicsItems.PlotItem.PlotItem import PlotItem
+from ..graphicsItems.ViewBox import ViewBox
+from ..widgets.PlotWidget import PlotWidget
+
+
+class MultiAxisPlotWidget(PlotWidget):
+    def __init__(self, **kargs):
+        """PlotWidget but with support for multi axis.
+        usage: you generally want to call in this order:
+        .addAxis(...) as many times as axes needed,
+        .addChart(...) as many times as charts needed,
+        and then .makeLayout() once all the needed elements have been added.
+        Also consider calling .update() after updating the chart data.
+        Refer to the "MultiAxisExample" example if needed"""
+        PlotWidget.__init__(self, enableMenu=False, **kargs)
+        # plotitem shortcut
+        self.pi = super().getPlotItem()
+        # default vb from plotItem shortcut
+        self.vb = self.pi.vb
+        # layout shortcut
+        self.layout = self.pi.layout
+        # hide default axis
+        for a in ["left", "bottom", "right", "top"]:
+            self.pi.hideAxis(a)
+        # CHARTS
+        self.axes = {}
+        self.charts = {}
+        self._signalConnectionsByChart = {}
+
+    def addAxis(self, name, *args, axis=None, **kwargs):
+        """Add a new axis (AxisItem) to the widget, will be shown and linked to a chart when used in addChart().
```

Comment:
The following docstring sphinx will be happy with:

```python
        """
        Add a new axis (AxisItem) to the widget, will be shown and linked to a 
        chart when used in addChart().

        Parameters
        ----------
        name : str
            The name associated with this axis item, used to store and recall
            the newly created axis. Also sets the AxisItem name parameter.
        chart : AxisItem
            The axis to be used. If left as None a new AxisItem will be created
            and it's name parameter set.
        args : iterable 
            arguments to be passed to the newly created AxisItem
        kwargs : dict 
            arguments to be passed to the newly created AxisItem.  Remember to 
            set the AxisItem orientation.

        Returns
        -------
        AxisItem
            The newly created AxisItem.
        """
```

Reply:
done